lib/ansible/modules/user.py

```diff
@@ -1899,8 +1913,10 @@ def modify_user(self):
                                 groups = set(current_groups).union(groups)
                                 groups_need_mod = True
                                 break
+                        self.groups_diff['after'] = list(set(current_groups) | groups)
                     else:
                         groups_need_mod = True
+                        self.groups_diff['after'] = list(groups)
```

Comment:
I think that these lists should all be sorted consistently for predictability. So it'd be `sorted(list(...))` in every place, which begs for a helper function, making it possible to reduce copy-paste.

Maybe made something like this:
```python
def _convert_to_sorted_list(unspecified_iterable):
    return sorted(list(unspecified_iterable))
```

Reply:
They are sorted for the user in main().